src/Workspaces/Core/Portable/Serialization/SerializerService.cs

```diff
@@ -180,55 +180,32 @@ public void Serialize(object value, ObjectWriter writer, SolutionReplicationCont
         }
     }
 
-    public T Deserialize<T>(WellKnownSynchronizationKind kind, ObjectReader reader, CancellationToken cancellationToken)
+    public object Deserialize(WellKnownSynchronizationKind kind, ObjectReader reader, CancellationToken cancellationToken)
```

Comment:
specializing this method was non-sensical.  we'd just instantiate it for N different kinds, all of which would have this major unnecessary switch in it.  This should just be a non-generic method (deserializing, after all, can return one of many different types), and the caller should appropriately cast to what they want, instead of trying to push teh cast into this.